packages/flutter/lib/src/material/text_field.dart

```diff
@@ -1230,7 +1243,7 @@ class _TextFieldState extends State<TextField> with RestorationMixin implements
         textSelectionControls ??= cupertinoTextSelectionHandleControls;
         paintCursorAboveText = true;
         cursorOpacityAnimates = true;
-        cursorColor = widget.cursorColor ?? selectionStyle.cursorColor ?? cupertinoTheme.primaryColor;
+        cursorColor = _getCursorColor(selectionStyle, theme);
```

Comment:
Should `theme` be `cupertinoTheme` here?

Reply:
Great catch, Added Cupertino Theme for Mac and Ios platform case